rclcpp/src/rclcpp/publisher.cpp

```diff
@@ -94,6 +95,13 @@ PublisherBase::~PublisherBase()
       rcl_get_error_string().str);
     rcl_reset_error();
   }
+
+  auto ipm = weak_ipm_.lock();
+  if (!ipm) {
+    // TODO(ivanpauno): should this raise an error?
```

Comment:
@ivanpauno well, the publisher isn't necessarily using intra-process communication. We need another flag to know whether this should log an error or not.

Reply:
As this is called in the destructor, it could happens that the ipm have already gone out of the scope first (I'm not sure, but it sounds reasonable). So, I don't think rising an error is a good idea.
If we want to raise an error, we need an extra flag.

Reply:
@ivanpauno Looks like here we should be using the `use_intra_process_` flag as well to know whether this should be treated as an error or not.